lib/geoengineer/resources/aws/api_gateway/aws_api_gateway_model.rb

```diff
@@ -10,10 +10,15 @@ class GeoEngineer::Resources::AwsApiGatewayModel < GeoEngineer::Resource
 
   validate -> { validate_required_attributes([:rest_api_id, :name, :content_type, :schema]) }
 
+  after :initialize, -> { self.rest_api_id = _rest_api.to_ref if _rest_api }
   after :initialize, -> { _terraform_id -> { NullObject.maybe(remote_resource)._terraform_id } }
-  after :initialize, -> { _geo_id -> { rand(36**20).to_s(36) } }
+  after :initialize, -> { _geo_id -> { "#{_rest_api._geo_id}::#{name}" } }
```

Comment:
What if `_rest_api` is nil? Your if statement on line 13 seems to suggest it might be nil.

Reply:
_geo_id is really important and should never be nil - it is the resource's primary identifier.

Reply:
Removed the implication that `_rest_api` can ever be nil. It will now raise an error if it ever is.

Reply:
I think thats a good idea - the docs here suggest it is a Required attribute:

https://www.terraform.io/docs/providers/aws/r/api_gateway_model.html